contentcuration/contentcuration/static/js/edit_channel/channel_set/util.js

```diff
@@ -0,0 +1,56 @@
+var Models = require('../models');
+const State = require("edit_channel/state");
+
+exports.loadChannelSetChannels = function(token) {
+  return new Promise(function(resolve, reject) {
+    $.ajax({
+      method:"GET",
+      url: window.Urls.get_channels_by_token(token),
+      success: resolve,
+      error: reject
+    });
+  })
+}
+
+exports.loadChannelList = function(url) {
+  return new Promise(function(resolve, reject) {
+    $.ajax({
+      method:"GET",
+      url: url,
+      success: resolve,
+      error: reject,
+      data: {
+        "published": true
+      }
+    });
+  })
+}
+
+exports.saveChannelsToSet = function(channelSetData, channels) {
+  return new Promise(function(resolve, reject) {
+    let channelSetModel = new Models.ChannelSetModel();
+    delete channelSetData.secret_token; // Don't save this
+
+    if(!channelSetData.editors || channelSetData.editors.length === 0) {
+      channelSetData.editors = [State.current_user.id];
+    }
+
+    // Need to save first to make sure token is available
+    channelSetModel.save(channelSetData, {
+      patch: true,
+      error: reject,
+      success: function(channelSet) {
+        $.ajax({
+          method:"POST",
+          url: window.Urls.save_token_to_channels(channelSet.get('secret_token').token),
+          success: function() {
+            channelSet.fetch({success: resolve});
+          },
+          error: reject,
+          data: JSON.stringify(_.pluck(channels, 'id'))
```

Comment:
By default jQuery will stringify the data parameter for you (http://api.jquery.com/jquery.ajax/), so generally don't need to explicitly call JSON.stringify unless you are doing custom stringification.

Reply:
Hmmm... I'm getting a `ValueError: No JSON object detected` error when I remove it. Is there something else I need to do with the endpoint? Or is it because it's a flat list rather than a dict?

Reply:
Hrm, fair enough, no harm no foul!